src/formatter/json_formatter.js

```diff
@@ -156,7 +156,14 @@ export default class JsonFormatter extends Formatter {
         data.result.duration = testStep.result.duration
       }
       if (status === Status.FAILED && exception) {
-        data.result.error_message = exception.stack || exception
+        if (_.isString(exception)) {
+          data.result.error_message = exception
+        } else {
+          let { message = '', stack = '' } = exception
+          message = _.includes(stack, message) ? '' : message + '\n'
+          data.result.error_message =
+            message + stack || JSON.stringify(exception)
```

Comment:
This is hard for me to follow. Thoughts on breaking out a function `stringifyException` and focusing on making it easy to understand:
```js
function stringifyException(exception) {
  // if is string 
  //   return it
  // else if has message or stack
  //   if stack includes message
  //     return stack
  //   else 
  //     return message newline stack
  // else 
  //   return stringified exception
}
```
The function could then go in a util file and be unit tested with some sample exceptions from the different browsers and non-error objects

Reply:
You are right, move this code to a function would be a better solution.
Any suggestion on where to put this utils file?
What about using /src/formatter/helpers/error_helpers.js?

Reply:
Sounds good to me